lntest/itest/lnd_test.go

```diff
@@ -9384,6 +9384,11 @@ func testAsyncPayments(net *lntest.NetworkHarness, t *harnessTest) {
 	// All payments have been sent, mark the finish time.
 	timeTaken := time.Since(now)
 
+	// Wait for the revocation to be received so alice no longer has
+	// pending htlcs listed. This is needed due to the fact that we
+	// now pipeline the settles.
+	time.Sleep(2 * time.Second)
```

Comment:
Seems like we could instead wrap the block below in a `WaitPredicate` variant? Also I don't follow why this is needed, things should be _faster_ now, not slower.

Reply:
Ah ok, faster in the prefix of the route, but we still need to do a full dance in the remainder.

Reply:
Done